### DISCUSSION TITLE
live-log examples questions

### DISCUSSION THREAD

**Discussion by guangnian1126** [cesanta/mongoose#2127] [Ask questions, get answers] [answered]
I try to change timer period from 1000 to 5 ,per second message count is 20 times,not 200 times ,why?

![image](https://user-images.githubusercontent.com/128374724/226540793-6dc1780c-2381-4cc2-a156-f561b9e19f4c.png)

**Comment by scaprile** [comment, marked as answer]
Because, as depicted in the tutorials and documentation, this is a single-threaded event-driven system. The timers run when the event manager runs, and if your event manager sleeps for 50ms, the smallest timer lapse you may have is 50ms, and 1/0.05 = 20
https://mongoose.ws/documentation/tutorials/webui-push/#backend-implementation-2 links to
http://mongoose.ws/documentation/tutorials/timers/#minimal-example , which states
> It is important to keep the event manager polling timeout smaller than the timer interval, because timers are checked at every polling cycle. If there are no network events, the poll function would sleep for the whole polling interval and if it is larger than the timer timeout, a timer invocation could be missed.
